test/gtest/uct/test_peer_failure.cc

```diff
@@ -583,19 +583,30 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure_keepalive, killed,
     ASSERT_UCS_OK(status);
     flush();
 
-    /* allow keepalive requests to complete */
+    /* Allow keepalive requests to complete */
     short_progress_loop();
 
-    /* we are still alive */
+    /* We are still alive */
     EXPECT_EQ(0, m_err_count);
 
     kill_receiver();
 
-    status = uct_ep_check(ep0(), 0, NULL);
-    ASSERT_UCS_OK(status);
-    flush();
+    unsigned retries = 1;
+    do {
+        status = uct_ep_check(ep0(), 0, NULL);
+        ASSERT_UCS_OK(status);
+        flush();
+
+        wait_for_flag(&m_err_count);
+        if (m_err_count != 0) {
+            break;
+        }
+
+        /* Progress to make sure all HW objects (e.g. DCT) are fully
```

Comment:
why kill_receiver() does not close the DCT?

Reply:
it closes, but RDMA_WRITE is succesful right after that.
it seems that closing DCT takes some more time to be fully destroyed in FW, so we are able to send data successfully.

Reply:
maybe try to flush / wait some time before kill_receiver to let DCI disconnect?
anyway seems to check more internally why it happens

Reply:
> maybe try to flush / wait some time before kill_receiver to let DCI disconnect?

do you mean *after* `kill_receiver`?

> anyway seems to check more internally why it happens

yes, it worth checking why it happens